portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -22,31 +22,41 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   
-  private ArrayList<String> quotes;
+  private ArrayList<String> comments;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    quotes = new ArrayList<>();
-    quotes.add("beep boop - ben lin");
-    quotes.add("They told me computers could only do arithmetic -Grace Hopper. I told them I cannot do arithmetic - Ben Lin");
-    quotes.add("hydrate or diedrate - Ben Lin");
-
+    if (comments == null) {
+        comments = new ArrayList<>();
+        comments.add("Beep Boop - Ben Lin");
+    }
+    
     // Convert the quotes to JSON using Gson library
-    String json = new Gson().toJson(quotes);
+    String json = new Gson().toJson(comments);
 
     // Send the JSON as the response
     response.setContentType("application/json;");
     response.getWriter().println(json);
     
   }
 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  
+    // Get the input from the form and add it to json on /data
+    String commentSubmission = request.getParameter("comment-submission");
+    comments.add(commentSubmission);
```

Comment:
If you get a post request before a get request request, comments will be null. You can mitigate this by initializing it in line 33: `private List<String> comments = new ArrayList<>()`, then you also won't need the null check in the get handler

Reply:
Implemented!